contracts/auction/ClockAuction.sol

```diff
@@ -41,6 +54,31 @@ contract ClockAuction is AuctionRelated {
     }
 
 
+    // TODO: in genesis auction, if tokenId is reserverd,
+    // you need to buy it with kr.
+    function _buyWithKTON(address _from, uint256 _tokenId, uint256 _valueInKTON) internal returns (uint256){
+        // Get a reference to the auction struct
+        Auction storage auction = tokenIdToAuction[_tokenId];
+        // require this is the first bid of auction
+        require(auction.lastBidder == 0x0, "It's not the first bid, KTON is not allowed");
```

Comment:
lastBidder is 0 does mean that this nft is genesis, we should use sender == genesisSender to answer whether it is geneis.